### DISCUSSION TITLE
Which ERT Modelling Forward Operator to use

### DISCUSSION THREAD

**Discussion by andpalacios** [gimli-org/pyGIMLi#545] [Q&A] [answered]
Hello, 

I'm here to ask for help/directions with two things: 

1. I would like to solve simulate a 2D ERT to then perform inversion. I am unsure abour which Modelling Class I should use with the pg.Inversion class. At the moment I run my ERT forward simulation using **pg.physics.ert.simulate** but I don't know how to indicate this as the forward operator inside the inversion (**fop**). 

_data = pg.load("modelled_ert.dat")
inv = pg.Inversion(fop=fop, verbose=True)
invkw = dict(dataVals=data['rhoa'], errorVals = data['err'], startModel=np.mean(data['rhoa']), lam=100, robustData = True)
tLog = pg.trans.TransLog()
### ERROR HERE --> itnv.modelTrans(tLog)
inv.setRegularization(cType=2)  # smoothing
result = inv.run(**invkw)_

2. I am having issues applying the pg.trans.TransLog to my data (rho app). Is this normal? 

Thank you very much in advance for your help! 

Best regards,
Andrea

**Comment by halbmy** [comment, marked as answer]
The synthetic simulation should be completely independent on inversion, so it is perfect to use `ert.simulate` for the first and setup a forward operator for the latter. As most examples/tutorials on ERT use the `ERTManager`, this one might help you: 
https://www.pygimli.org/_examples_auto/6_inversion/plot_5_ert_with_priors.html
as it uses an own inversion instance (in section Joint Inversion). 

So you can either
* setup a manager and work with `mgr.inv` as inversion instance
* setup a manager and work with `mgr.fop` as forward operator, or
* setup `ert.Modelling` by yourself:
```
from pygimli.physics import ert
fop = ert.Modelling()
fop.setData(data)
fop.setMesh(mesh)
```

The above example also shows you how to use the transformations:
```
inv.modelTrans = tLog
inv.dataTrans = tLog
```

**Comment by halbmy** [comment]
P.S.: In the title you mention `RInversion`. This core inversion should not be used but `pg.Inversion` or `pg.frameworks.MeshInversion`
